### PR TITLE
Removed message_record, replace with in_transit_message_queue

### PR DESCRIPTION
**Pull request by chanijjani**
- This PR follows the relevant and the latest PR in reactor-c.
 (https://github.com/lf-lang/reactor-c/commit/2f1c9df23fa2ab0ae38accfd041fa15201ca8b9b).

### CODE REVIEW

**Comment by chanijjani** [comment]
This PR let Rust RTI pass the [AfterDelay.lf](https://github.com/lf-lang/lingua-franca/blob/master/test/C/src/federated/AfterDelays.lf) test (and maybe other federated tests).
This PR needs to be rebased after https://github.com/lf-lang/rust-rti/pull/44 is merged.

**Review by edwardalee — COMMENTED** [review]
I'm not qualified to comment on the Rust code, but one concern is that the file `pqueue_tag.rs` says:
```
 * @brief Priority queue that uses tags for sorting.
```
But this does not seem to be what is implemented in this file.  It seems that what is implemented is very specifically the in-transit message queue of the RTI.  Would it make more sense to actually have a generic pqueue_tag and then use it in the RTI? In that case, this file would have no dependencies on RTI or federate code.

**Comment by chanijjani** [comment]
@edwardalee Professor, thank you for your time and valuable feedback. I will work on creating a more generic pqueue_tag implementation.

**Comment by lhstrh** [comment]
> ```
>  * @brief Priority queue that uses tags for sorting.
> ```
> 
> But this does not seem to be what is implemented in this file. It seems that what is implemented is very specifically the in-transit message queue of the RTI. Would it make more sense to actually have a generic pqueue_tag and then use it in the RTI? In that case, this file would have no dependencies on RTI or federate code.

@edwardalee: A generic priority queue already exists in the std library and that is precisely what is being used here (see line 11). So rather than create a generic pqueue, I think your concern will be better addressed by making the comments more descriptive...

File: rust/rti/src/federate_info.rs [hunk 1]
```diff
@@ -8,16 +7,11 @@ use crate::message_record::message_record::InTransitMessageRecordQueue;
  * @author Chanhee Lee (chanheel@asu.edu)
  * @author Hokeun Kim (hokeun@asu.edu)
  * @copyright (c) 2020-2024, The University of California at Berkeley
- * License in [BSD 2-clause](..)
+ * License in [BSD 2-clause](https://github.com/lf-lang/reactor-c/blob/main/LICENSE.md)
  * @brief Declarations for runtime infrastructure (RTI) for distributed Lingua Franca programs.
- * This file extends rti_common.h with RTI features that are specific to federations and are not
+ * This file extends enclave.h with RTI features that are specific to federations and are not
```
Comment (lhstrh) [inline rust/rti/src/federate_info.rs:12]: This comment is no longer accurate.

**Review by lhstrh — COMMENTED** [review]
I'm not the right person to review this code, but the changes look like substantial improvements to me...

**Comment by lhstrh** [comment]
For other reviewers: this PR has a lot of overlap with #44, so I recommend reviewing (and merging) that PR first. Then, the diff will be a lot smaller.

File: rust/rti/src/federate_info.rs (hunk 1 above)
Comment (chanijjani) [reply rust/rti/src/federate_info.rs]: Thanks. Modified!

**Comment by chanijjani** [comment]
@lhstrh, you're correct. After https://github.com/lf-lang/rust-rti/pull/44 is merged, this PR needs to be rebased. Sorry for any confusion caused.

**Review by hokeun — APPROVED** [review]
👍

**Pull request merged**